### ISSUE TITLE
t.i.utils process functions should have a default cwd of None, not '.'

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3159]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#3159|
|Type|defect|
|Created|2008-04-03 16:52:28Z|
|Branch|https://github.com/twisted/twisted/tree/utils-none-cwd-3159|

This causes problems when the process is being run in a directory that it doesn't have permissions to chdir to (most commonly this happens when you're passing the uid option). These processes would run fine otherwise. Passing None as cwd to spawnProcess would avoid the chdir entirely, which would avoid this problem. I can't think of any backwards incompatibility that this would introduce.

The exception this causes is something like:

```
  File "/usr/lib/python2.5/site-packages/twisted/internet/process.py", line 515, in _execChild
    os.chdir(path)
OSError: [Errno 13] Permission denied: '.'
```

<details><summary>Searchable metadata</summary>

```
trac-id__3159 3159
type__defect defect
reporter__radix radix
priority__normal normal
milestone__ 
branch__branches_utils_none_cwd_3159 branches/utils-none-cwd-3159
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1207241548000000 1207241548000000
changetime__1214675345000000 1214675345000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

Man, your applications are crazy.

But I think I agree.  +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24105]) Branching to 'utils-none-cwd-3159'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

build results:

http://buildbot.twistedmatrix.com/boxes-supported?num_builds=1&branch=/branches/utils-none-cwd-3159

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Ready for review.

There were a couple revisions after that build, but nothing major.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [24144]) Merge utils-none-cwd-3159

Author: exarkun
Reviewer: therve
Fixes: [#3159](https://github.com/twisted/twisted/issues/3159)

Change the default for the `path` parameter to the process utility
functions in `twisted.internet.utils` from `.` to `None`.  Each
of these results in the child's working directory being the same
as the parent's, but `None` avoids the attempt to actually change
to the directory.  This is a no-op, except when there is a permission
error which results in the child process not being able to start.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
